src/bundle.c

```diff
@@ -855,10 +856,20 @@ static int install_bundles(struct list *bundles, struct list **subs, int current
 	run_scripts(false);
 
 	ret = 0;
-	fprintf(stderr, "Bundle(s) installation done.\n");
 	print_time_stats(&times);
 
 out:
+	bundles_installed = list_len(to_install_bundles);
```

Comment:
This isn't going to work. to_install_bundles contains bundles on bundles list + dependencies. So you may have a situation where bundles_installed > bundles_requested and then bundles_failed < 0. And in this case if a bundle fails to install we'll have bundles_failed <= 0 and a success message will be printed.

In a system without any desktop bundles:

    # swupd bundle-add desktop-apps-extras
    list_len(to_install_bundles) == 6
    list_len(bundles) == 1

Reply:
yep, you are right. I'll solve this issue in another way so it is more reliable. I'm going to fix this commit.